### DISCUSSION TITLE
Drag edges for alternate route (step or smoothstep)

### DISCUSSION THREAD

**Discussion by rogeru** [bcakmakoglu/vue-flow#679] [Ideas]
With a more complicated flow the edges cross eachother quite often. If step or smoothstep edge lines would be dragable, the flow could be made look much simpler by reducing edges crossing eachother.

Is that something that's planned? If not, I may take a look on how to enhance those edges with that capability. And suggestion where to start?

For example the "timeout" out handle could be moved more right so it does not cover the disconnect node.
<img width="511" alt="image" src="https://user-images.githubusercontent.com/1393401/218863042-6fd37c1b-fae5-4653-b730-ba282e86ba4d.png">

**Comment by bcakmakoglu** [comment]
Draggable or edges with custom control points are something I want to work on at some point but I haven’t had the time to do so :/
If you want to try working on it you could simply create a separate package for this custom edge (similar to the path finding edge in this repo).
As to pointers, might be good to read up on control points for edges or rather svg lines 😅

**Comment by snoozbuster** [comment]
I also have found that I need something like this. Does anything similar exist for react-flow that we might be able to learn from? I did some googling but I couldn't find it - maybe I don't know the proper terms for it.

**Reply by snoozbuster** [reply]
[maxgraph](https://jgraph.github.io/mxgraph/javascript/examples/grapheditor/www/index.html) has a pretty robust stepped edge editor. some of the code exists [here](https://github.com/maxGraph/maxGraph/blob/development/packages/core/src/view/handler/ElbowEdgeHandler.ts) but other parts of it seem to exist in other handler files in that same folder, like `EdgeHandler` and `EdgeSegmentHandler`. the actual UI events are wired up to these functions and data separately but I think most of the complex calculation logic lives in these files and might be a good starting point.

**Reply by bcakmakoglu** [reply]
Thanks for the reference, I'll look into it and see if I can take some ideas from there, though I'm currently a bit short on time :/
